cmd/podman/main.go

```diff
@@ -40,8 +43,32 @@ func main() {
 
 	if filepath.Base(os.Args[0]) == registry.PodmanSh ||
 		(len(os.Args[0]) > 0 && filepath.Base(os.Args[0][1:]) == registry.PodmanSh) {
+		if !rootless.IsRootless() {
+			fmt.Println("podmansh not supported for root user")
+			os.Exit(125)
+		}
+		timeout := "10"
+
+		usr, err := user.LookupId(strconv.Itoa(rootless.GetRootlessUID()))
```

Comment:
I don't think that'll work when running inside the user namespace.

Please add code comments.  I am not entirely sure what the code's intention is.  A comment would be helpful.